src/vivarium_nih_us_cvd/components/disease.py

```diff
@@ -43,41 +43,132 @@ def IschemicStroke():
     )
 
 
-def MyocardialInfarction():
-    susceptible = SusceptibleState(models.MYOCARDIAL_INFARCTION_MODEL_NAME)
-    data_funcs = {"dwell_time": lambda *args: pd.Timedelta(days=28)}
+def IschemicHeartDiseaseAndHeartFailure():
+    susceptible = SusceptibleState(models.ISCHEMIC_HEART_DISEASE_AND_HEART_FAILURE_MODEL_NAME)
+    acute_mi_dwell_data_funcs = {"dwell_time": lambda *args: pd.Timedelta(days=28)}
+    acute_mi_and_hf_data_funcs = {
+        "dwell_time": lambda *args: pd.Timedelta(days=28),
+        "disability_weight": lambda _, builder: builder.data.load(
+            data_keys.IHD_AND_HF.DISABILITY_WEIGHT_ACUTE_MI
+        ),
+        "excess_mortality_rate": lambda _, builder: builder.data.load(
+            data_keys.IHD_AND_HF.EMR_ACUTE_MI
+        ),
+    }
+    heart_failure_emr_data_funcs = {
+        "excess_mortality_rate": lambda _, builder: builder.data.load(
+            data_keys.IHD_AND_HF.EMR_HF
+        )
+    }
+    # states without heart failure
     acute_myocardial_infarction = DiseaseState(
         models.ACUTE_MYOCARDIAL_INFARCTION_STATE_NAME,
         cause_type="cause",
-        get_data_functions=data_funcs,
+        get_data_functions=acute_mi_dwell_data_funcs,
     )
     post_myocardial_infarction = DiseaseState(
         models.POST_MYOCARDIAL_INFARCTION_STATE_NAME,
         cause_type="cause",
     )
 
-    susceptible.allow_self_transitions()
-    data_funcs = {
+    # states with heart failure
+    heart_failure_from_ihd = DiseaseState(
+        models.HEART_FAILURE_FROM_ISCHEMIC_HEART_DISEASE_STATE_NAME,
+        cause_type="cause",
+        get_data_functions=heart_failure_emr_data_funcs,
+    )
+    residual_heart_failure = DiseaseState(
+        models.HEART_FAILURE_RESIDUAL_STATE_NAME,
+        cause_type="cause",
+        get_data_functions=heart_failure_emr_data_funcs,
```

Comment:
Why does this get the same data funcs as HF_IHD?

Reply:
Both heart failure states have the same EMR.